stan/math/prim/arr/fun/promote_elements.hpp

```diff
@@ -3,6 +3,7 @@
 
 #include <stan/math/prim/scal/fun/promote_elements.hpp>
 #include <vector>
+#include <cstddef>
```

Comment:
Maybe then it's an issue of Ubuntu system include files? I got the following error:


        g++ -w -Wfatal-errors -I . -isystem lib/eigen_3.2.9 -isystem lib/boost_1.62.0 -isystemlib/cvodes_2.9.0/include -Wall -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DNO_FPRINTF_OUTPUT -pipe -Wno-unused-function -Wno-uninitialized -Wno-c++11-long-long   -c -O0 -include stan/math/prim/arr/fun/promote_elements.hpp -o /dev/null test/dummy.cpp
        In file included from <command-line>:0:0:
        ./stan/math/prim/arr/fun/promote_elements.hpp: In static member function ‘static std::vector<T> stan::math::promote_elements<std::vector<T>, std::vector<S> >::promote(const std::vector<S>&)’:
        ./stan/math/prim/arr/fun/promote_elements.hpp:30:14: error: ‘size_t’ was not declared in this scope
                 for (size_t i = 0; i < u.size(); ++i)
                      ^~~~~~
        compilation terminated due to -Wfatal-errors.
        ****



        clang++ -w -Wfatal-errors -I . -isystem lib/eigen_3.2.9 -isystem lib/boost_1.62.0 -isystemlib/cvodes_2.9.0/include -Wall -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DNO_FPRINTF_OUTPUT -pipe -Wno-unused-function -Wno-uninitialized -Wno-c++11-long-long   -c -O0 -include stan/math/prim/arr/fun/promote_elements.hpp -o /dev/null test/dummy.cpp
        In file included from <built-in>:325:
        In file included from <command line>:5:
        ./stan/math/prim/arr/fun/promote_elements.hpp:30:14: fatal error: unknown type
              name 'size_t'; did you mean '__gnu_cxx::size_t'?
                for (size_t i = 0; i < u.size(); ++i)
                     ^~~~~~
                     __gnu_cxx::size_t
        /usr/bin/../lib/gcc/x86_64-linux-gnu/6.2.0/../../../../include/c++/6.2.0/ext/new_allocator.h:44:14: note: 
              '__gnu_cxx::size_t' declared here
          using std::size_t;
                     ^
        1 error generated.